bindings/node.js/kzg.ts

```diff
@@ -0,0 +1,116 @@
+// @ts-expect-error
+import bindings from "bindings";
+
+/**
+ * The public interface of this module exposes the functions as specified by
+ * https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/polynomial-commitments.md#kzg
+ */
+
+export type BLSFieldElement = Uint8Array; // 32 bytes
+export type KZGProof = Uint8Array; // 48 bytes
+export type KZGCommitment = Uint8Array; // 48 bytes
+export type Blob = Uint8Array; // 4096 * 32 bytes
+
+type SetupHandle = Object;
+
+// The C++ native addon interface
+type KZG = {
+  FIELD_ELEMENTS_PER_BLOB: number;
+  BYTES_PER_FIELD: number;
+
+  loadTrustedSetup: (filePath: string) => SetupHandle;
+
+  freeTrustedSetup: (setupHandle: SetupHandle) => void;
+
+  blobToKzgCommitment: (blob: Blob, setupHandle: SetupHandle) => KZGCommitment;
+
+  computeAggregateKzgProof: (
+    blobs: Blob[],
+    setupHandle: SetupHandle
+  ) => KZGProof;
+
+  verifyAggregateKzgProof: (
+    blobs: Blob[],
+    expectedKzgCommitments: KZGCommitment[],
+    kzgAggregatedProof: KZGProof,
+    setupHandle: SetupHandle
+  ) => boolean;
+
+  verifyKzgProof: (
+    polynomialKzg: KZGCommitment,
+    z: BLSFieldElement,
+    y: BLSFieldElement,
+    kzgProof: KZGProof,
+    setupHandle: SetupHandle
+  ) => boolean;
+};
+
+const kzg: KZG = bindings("kzg.node");
```

Comment:
Here's the import of the NodeJS native addon.

This file exports types and export function wrappers for the KZG module.